lib/oga/xml/parser.rll

```diff
@@ -245,7 +245,7 @@ string_body
   def reset
     @line = 1
 
-    @lexer.reset
+    @lexer.reset_native
   end
```

Comment:
@sferik If the code of the `reset` method is moved into the `initialize` method I think it should work. Which tests were failing?